internal/app/add/add.go

```diff
@@ -1,90 +1,82 @@
 	if globalFlag {
-		errAddTodoAndAssociateTo := addTodoAndAssociateTo("", args)
-		if errAddTodoAndAssociateTo != nil {
-			return fmt.Errorf("addTodoAndAssociateTo: %w", errAddTodoAndAssociateTo)
+		if err := addTodoAndAssociateTo("", args); err != nil {
+			return fmt.Errorf("addTodoAndAssociateTo: %w", err)
 		}
 		return nil
 	}
 
-	homeRemCurrProExDir, errHomeRemCurrProExDir := shared.GetHomeRemovedWorkingDir()
-	if errHomeRemCurrProExDir != nil {
-		return fmt.Errorf("helper.GetHomeRemovedCurrentProgramExecutionDirectory: %w", errHomeRemCurrProExDir)
+	dir, err := shared.GetHomeRemovedWorkingDir()
+	if err != nil {
+		return fmt.Errorf("helper.GetHomeRemovedCurrentProgramExecutionDirectory: %w", err)
 	}
 
-	if errAddTodoAndAssociateTo := addTodoAndAssociateTo(homeRemCurrProExDir, args); errAddTodoAndAssociateTo != nil {
-		return fmt.Errorf("addTodoAndAssociateTo: %w", errAddTodoAndAssociateTo)
+	if err := addTodoAndAssociateTo(dir, args); err != nil {
+		return fmt.Errorf("addTodoAndAssociateTo: %w", err)
 	}
 
 	return nil
 }
 
 func addTodoAndAssociateTo(directory string, todoListString []string) error {
 	// Find the record in the record file
-	recordItems, errGetRecordFileContents := record.GetFileContents()
-	if errGetRecordFileContents != nil {
-		return fmt.Errorf("helper.GetRecordFileContents: %w", errGetRecordFileContents)
+	recordItems, err := record.GetFileContents()
+	if err != nil {
+		return fmt.Errorf("helper.GetRecordFileContents: %w", err)
 	}
 
 	idx := slices.IndexFunc(recordItems, func(item record.RecordEntity) bool {
 		return item.Path == directory
 	})
 
-	dataFolder, errGetDataFolder := data.GetFolder()
-	if errGetDataFolder != nil {
-		return fmt.Errorf("helper.GetDataFolder: %w", errGetDataFolder)
+	dataFolder, err := data.GetFolder()
+	if err != nil {
+		return fmt.Errorf("helper.GetDataFolder: %w", err)
 	}
 
-	var currentDirectoryRecord *record.RecordEntity
+	var current *record.RecordEntity
 	if idx == -1 {
-		tempCurrentDirectoryRecord, errCreateNewRecord := record.CreateNewRecord(directory)
-		if errCreateNewRecord != nil {
-			return fmt.Errorf("helper.CreateNewRecord: %w", errCreateNewRecord)
+		// Record was not found.
+		// Create a new one and add it to recordItems.
+
+		tmp, err := record.CreateNewRecord(directory)
+		if err != nil {
+			return fmt.Errorf("helper.CreateNewRecord: %w", err)
 		}
-		currentDirectoryRecord = &tempCurrentDirectoryRecord
-		recordItems = append(recordItems, *currentDirectoryRecord)
+		current = &tmp
+		recordItems = append(recordItems, tmp)
 		if err := record.SetFileContents(recordItems); err != nil {
 			return fmt.Errorf("helper.SetRecordFileContents: %w", err)
 		}
 	} else {
-		currentDirectoryRecord = &recordItems[idx]
+		current = &recordItems[idx]
 	}
 
 	// Read the file, it will exist if it reaches here
-	dataFileFullPath := dataFolder + string(os.PathSeparator) + currentDirectoryRecord.DataFileName
-	_, errStat := os.Stat(dataFileFullPath)
```

Comment:
This call to `os.Stat` doesn't seem to do anything. If the file doesn't exist or there's a problem reading it, that'll show up in the subsequent call to `GetTodoFromFile`.